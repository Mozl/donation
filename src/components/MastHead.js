import React from 'react';
import styled from 'styled-components';

const Header = styled.div`
  display: flex;
  flex-direction: row;
  position: sticky;
  top: 0;
  height: 4rem;
  width: 100%;
  background-color: white;
  align-items: center;
  justify-content: space-between;

  @media only screen and (max-width: 600px) {
    margin-top: 0.5rem;
    flex-direction: column;
  }
`;

const ButtonWrapper = styled.div`
  align-items: end;
`;

const Button = styled.button`
  border: 0.1rem solid black;
  margin: 0.8rem;
  border-radius: 10px;
`;

class MastHead extends React.Component {
  render() {
    return (
      <Header>
        <img
          alt="Just Giving"
          src="https://www.jg-cdn.com/chrome/3e7a4dc9d866fda15331414a9072da97.svg"
        />
        <ButtonWrapper>
          <Button>About Us</Button>
        </ButtonWrapper>
      </Header>
    );
  }
}
export default MastHead;
