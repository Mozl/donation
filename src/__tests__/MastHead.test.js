import React from 'react';
import ReactDOM from 'react-dom';
import MastHead from '../components/MastHead';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MastHead />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<MastHead />);
  expect(tree).toMatchSnapshot();
});

it('renders logo image', () => {
  const masthead = shallow(<MastHead />);
  const logoImage =
    'https://www.jg-cdn.com/chrome/3e7a4dc9d866fda15331414a9072da97.svg';
  expect(masthead.find('img').prop('src')).toEqual(logoImage);
});
