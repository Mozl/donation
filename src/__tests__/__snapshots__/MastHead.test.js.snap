// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdVaJa giOLUk"
>
  <img
    alt="Just Giving"
    src="https://www.jg-cdn.com/chrome/3e7a4dc9d866fda15331414a9072da97.svg"
  />
  <div
    className="sc-bwzfXH dRBnFW"
  >
    <button
      className="sc-htpNat eJPqvy"
    >
      About Us
    </button>
  </div>
</div>
`;
